staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -360,3 +378,22 @@ func (s *WatchServer) HandleWS(ws *websocket.Conn) {
 		}
 	}
 }
+
+func recordWatchListLatency(ctx context.Context, reqScope *RequestScope) {
+	requestReceivedTimestamp, ok := request.ReceivedTimestampFrom(ctx)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("unable to measure watchlist latency because no received ts found in the ctx, gvr: %s", reqScope.Resource))
+		return
+	}
+	requestInfo, ok := request.RequestInfoFrom(ctx)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("unable to measure watchlist latency because no req info found in the ctx, gvr: %s", reqScope.Resource))
+		return
+	}
+	// TODO(p0lyn0mial): figure out how to get ns and name for watch requests
```

Comment:
Why? We don't need name/namespace - we are only reporting it per scope...

Reply:
Yes, this is required for the `scope`. I'm pretty sure it will work most of the time. 
Except one case mentioned in https://github.com/kubernetes/kubernetes/pull/120520#discussion_r1329653556

The worst case scenario would be that we end up with an empty `scope`

Reply:
OK - I see what you're saying now.

If you could open a separate PR with tests that Jordan mentioned and a fix for the missing namespace?

Reply:
yes, I was planning to add support for it in https://github.com/kubernetes/perf-tests/pull/2316